### PR TITLE
feat(Builder): Adds node to send messages to discord webhook

### PR DESCRIPTION
**Pull request by Bentlybro**
### Background

This adds a node that lets users sends messages into discord via discord webhooks

### Changes 🏗️

image

image

### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [x] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [ ] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [ ] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

### CODE REVIEW

File: rnd/autogpt_server/autogpt_server/blocks/discordsendmessage.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import requests
+from autogpt_server.data.block import Block, BlockSchema, BlockOutput
+
+
+class discordsendmessage(Block):
```
Comment (Swiftyos) [inline rnd/autogpt_server/autogpt_server/blocks/discordsendmessage.py]: Naming convention in python is `DiscordSendMessage`

File: rnd/autogpt_server/autogpt_server/blocks/discordsendmessage.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import requests
```
Comment (Swiftyos) [inline rnd/autogpt_server/autogpt_server/blocks/discordsendmessage.py]: Probably want to call the file just `discord.py` as we can add more blocks into it in time.

File: rnd/autogpt_server/autogpt_server/blocks/discordsendmessage.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    class Output(BlockSchema):
+        status: str
+
+    def __init__(self):
+        super().__init__(
+            id="b3a9c1f2-5d4e-47b3-9c4e-2b6e4d2c4f3e",
+            input_schema=discordsendmessage.Input,
+            output_schema=discordsendmessage.Output,
+            test_input={
+                "webhook_url": "https://discord.com/api/webhooks/your_webhook_url",
```
Comment (Swiftyos) [inline rnd/autogpt_server/autogpt_server/blocks/discordsendmessage.py]: Are there any credentials required?

**Review by Swiftyos — CHANGES_REQUESTED** [review]

File: rnd/autogpt_server/autogpt_server/blocks/discordsendmessage.py (hunk 3 above)
Comment (Bentlybro) [reply rnd/autogpt_server/autogpt_server/blocks/discordsendmessage.py]: Not technically, they are part of the webhook url

File: rnd/autogpt_server/autogpt_server/blocks/__init__.py [hunk 4]
```diff
@@ -1,10 +1,11 @@
 from autogpt_server.blocks import sample, reddit, text, ai
```
Comment (Swiftyos) [inline rnd/autogpt_server/autogpt_server/blocks/__init__.py]: You need to add discord to this line

File: rnd/autogpt_server/autogpt_server/blocks/__init__.py [hunk 5]
```diff
@@ -1,10 +1,11 @@
...
+from autogpt_server.blocks.discord import DiscordSendMessage
 from autogpt_server.data.block import Block
 
 AVAILABLE_BLOCKS = {
     block.id: block
     for block in [v() for v in Block.__subclasses__()]
 }
 
-__all__ = ["ai", "sample", "reddit", "text", "AVAILABLE_BLOCKS", "GetWikipediaSummary"]
+__all__ = ["ai", "sample", "reddit", "text", "AVAILABLE_BLOCKS", "GetWikipediaSummary", "DiscordSendMessage"]
```
Comment (Swiftyos) [inline rnd/autogpt_server/autogpt_server/blocks/__init__.py]: Then add discord to this line too

Comment (Swiftyos) [reply rnd/autogpt_server/autogpt_server/blocks/__init__.py]: Sorry I missed this on you Wikipedia PR, you need to do the same for wikipedia

**Review by Swiftyos — CHANGES_REQUESTED** [review]

File: rnd/autogpt_server/autogpt_server/blocks/__init__.py (hunk 5 above)
Comment (Bentlybro) [reply rnd/autogpt_server/autogpt_server/blocks/__init__.py]: @Swiftyos think i fixed it now, i tested and it still works

**Review by Swiftyos — APPROVED** [review]

**Pull request merged**